tuna/machine.py

```diff
@@ -377,25 +432,15 @@ def get_gpu_clock(self, gpu_num=0):
 
     return sclk, mclk
 
-  def set_gpu_clock(self):
-    """set gpu clock level with rocm-smi"""
-    cnx = self.connect()
-
-    if self.sclk is not None:
-      sclk_cmd = f'{ROCMSMI} --setsclk {self.sclk}'
-      cnx.exec_command(sclk_cmd)
-    if self.mclk is not None:
-      mclk_cmd = f'{ROCMSMI} --setmclk {self.mclk}'
-      cnx.exec_command(mclk_cmd)
-
-  def restart_server(self, wait=True):
+  def restart_server(self, wait: bool = True) -> bool:
     """restart the machine"""
     logger = self.get_logger()
 
     logger.warning('Sending remote reboot command')
     if self.ipmi_ip is not None and not self.ipmi_inaccessible:
       logger.info('Using IPMI to reboot machine')
-      self.mmi.restart_server()
+      if self.mmi is not None:
```

Comment:
remove this condition and expand the above one like i suggested

Reply:
check is removed and added in line 440